cmd/nvidia-ctk/cdi/generate/generate.go

```diff
@@ -276,7 +283,17 @@ func (m command) generateSpec(opts *options) (spec.Interface, error) {
 		nvcdi.WithCSVIgnorePatterns(opts.csv.ignorePatterns.Value()),
 		// We set the following to allow for dependency injection:
 		nvcdi.WithNvmlLib(opts.nvmllib),
-	)
+	}
+
+	if len(opts.disableHooks.Value()) > 0 {
+		for _, hook := range opts.disableHooks.Value() {
+			for _, hookName := range nvcdi.NewHookName(hook) {
+				initOpts = append(initOpts, nvcdi.WithDisabledHook(hookName))
+			}
+		}
+	}
```

Comment:
The `len` check is not required. One can always iterate over an empty slice.
```suggestion
	for _, hook := range opts.disableHooks.Value() {
		initOpts = append(initOpts, nvcdi.WithDisabledHook(hook))
	}
```

(not that I have updated `WithDisabledHook` to:
```
// WithDisabledHook allows specific hooks to the disabled.
// This option can be specified multiple times for each hook.
func WithDisabledHook[T string | HookName](hook T) Option {
	return func(o *nvcdilib) {
		if o.disabledHooks == nil {
			o.disabledHooks = make(map[HookName]bool)
		}
		o.disabledHooks[HookName(hook)] = true
	}
}
```
to allow it to accept both `string` and `HookName` arguments.

Reply:
Suggestion taken